### ISSUE TITLE
[Web] `NavigationRail` should allow customizing link handling

### ISSUE THREAD

**Issue by muriloventuroso** [flutter/flutter#123772]
I'm developing an application with Flutter Web and I use NavigationRail to design the side menu. I would like users to be able to open some of these menus in a new tab or window using the right mouse button (just like you can on a conventional website).

In other parts of the system I can accomplish this using the url_launcher's Link widget. However, in NavigationRail I didn't find a way to do that.

I believe it would be interesting to incorporate this function into NavigationRail, as it is very useful for applications with Flutter Web.

**Comment by iapicca** [comment]
@muriloventuroso 
isn't [NavigationRail](https://api.flutter.dev/flutter/material/NavigationRail-class.html) supposed to "navigate"?

> A Material Design widget that is meant to be displayed at the left or right of an app to navigate between a small number of views, typically between three and five.

(from docs)

if you expect a navigation rail to something else maybe is better to build a custom widget for that

**Comment by muriloventuroso** [comment]
Yes, but on web systems the user may want to open that particular view in a new tab.

**Comment by huycozy** [comment]
>just like you can on a conventional website

@muriloventuroso Does your proposal look like this?

![Screenshot 2023-03-31 at 14 27 11](https://user-images.githubusercontent.com/104349824/229053054-219d950a-1e83-469b-a56e-05e80ff07198.png)

**Comment by muriloventuroso** [comment]
Exactly that way.

**Comment by huycozy** [comment]
Although I have the same thoughts as @iapicca that you can handle by yourself, it's helpful if NavigationRail has this feature. Labeling the issue as a proposal for more thought. Thank you!

**Comment by mdebbar** [comment]
This is a framework-side request, not a web request. So I'm removing the `platform-web` label.

We basically need to give users more control over what gets built for each rail destination.

FYI @goderbauer @chunhtai

**Comment by chunhtai** [comment]
I am not sure who is the owner of navigation rail maybe @QuncCccccc ?

**Comment by HansMuller** [comment]
We could extend NavigationRailDestination so that one could override the default onTap response so that one could launch a menu rather than just delegating to the NavigationRails onDestinationSelection callback. We don't have plans to do that in the near term although we'd accept PRs that add the feature.
